src/snowcli/cli/nativeapp/artifacts.py

```diff
@@ -47,6 +52,29 @@ def __init__(self, path: Path):
         self.path = path
 
 
+class ManifestYmlFileNotFoundError(ClickException):
```

Comment:
Said this before, but IMHO we have way too many of these exceptions. They're oddly specific. We shouldn't need to add 2-3 exception types for each change.

Reply:
I don't understand what "important enough to warrant its own exception" means. Can you explain? We have, in the native apps module, 18 exception types if I count correctly. I'm worried we're creating bloat without any benefit.